cluster-autoscaler/processors/nodes/post_filtering_processor.go

```diff
@@ -17,28 +17,68 @@ limitations under the License.
 package nodes
 
 import (
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
 	"k8s.io/autoscaler/cluster-autoscaler/context"
 	"k8s.io/autoscaler/cluster-autoscaler/simulator"
+	klog "k8s.io/klog/v2"
 )
 
 // PostFilteringScaleDownNodeProcessor selects first maxCount nodes (if possible) to be removed
 type PostFilteringScaleDownNodeProcessor struct {
 }
 
 // GetNodesToRemove selects up to maxCount nodes for deletion, by selecting a first maxCount candidates
-func (n *PostFilteringScaleDownNodeProcessor) GetNodesToRemove(ctx *context.AutoscalingContext, candidates []simulator.NodeToBeRemoved, maxCount int) []simulator.NodeToBeRemoved {
+func (p *PostFilteringScaleDownNodeProcessor) GetNodesToRemove(ctx *context.AutoscalingContext, candidates []simulator.NodeToBeRemoved, maxCount int) []simulator.NodeToBeRemoved {
 	end := len(candidates)
 	if len(candidates) > maxCount {
 		end = maxCount
 	}
-	return candidates[:end]
+	return p.filterOutIncompleteAtomicNodeGroups(ctx, candidates[:end])
 }
 
 // CleanUp is called at CA termination
-func (n *PostFilteringScaleDownNodeProcessor) CleanUp() {
+func (p *PostFilteringScaleDownNodeProcessor) CleanUp() {
 }
 
 // NewPostFilteringScaleDownNodeProcessor returns a new PostFilteringScaleDownNodeProcessor
 func NewPostFilteringScaleDownNodeProcessor() *PostFilteringScaleDownNodeProcessor {
 	return &PostFilteringScaleDownNodeProcessor{}
 }
+
+func (p *PostFilteringScaleDownNodeProcessor) filterOutIncompleteAtomicNodeGroups(ctx *context.AutoscalingContext, nodes []simulator.NodeToBeRemoved) []simulator.NodeToBeRemoved {
+	nodesByGroup := map[cloudprovider.NodeGroup][]simulator.NodeToBeRemoved{}
+	result := []simulator.NodeToBeRemoved{}
+	for _, node := range nodes {
+		nodeGroup, err := ctx.CloudProvider.NodeGroupForNode(node.Node)
+		if err != nil {
+			klog.Errorf("Node %v will not scale down, failed to get node info: %s", node.Node.Name, err)
+			continue
+		}
+		autoscalingOptions, err := nodeGroup.GetOptions(ctx.NodeGroupDefaults)
+		if err != nil {
+			klog.Errorf("Failed to get autoscaling options for node group %s: %v", nodeGroup.Id(), err)
+			continue
+		}
```

Comment:
Cloud Providers were previously free to return `nil, cloudprovider.ErrNotImplemented` for `NodeGroup.GetOptions()`. All 5 cloudprovider NodeGroups that I spot checked return this, including `Hetzner` which I am using.

With the current `master` branch I get an error from these lines when cluster-autoscaler tries to scale down my node group. Causing `cluster-autoscaler` to never scale down:

```
E0815 10:24:08.433379       1 post_filtering_processor.go:59] Failed to get autoscaling options for node group pool1: Not implemented
```

I think we can add an explicit check for `cloudprovider.ErrNotImplemented` here and consider it for the standard scale down.